bot.py

```diff
@@ -1,107 +1,120 @@
-                + f"There are {self.queue.qsize()} submissions in the queue)"
-            )
-        except Exception:
-            logger.exception("Error while queueing submission")
-            await msg.reply("Error handling your message", embed=constants.HELP_REPLY)
-        return
-
-    async def handle_help(self, msg):
-        await msg.reply(embed=constants.HELP_REPLY)
-        return
-
-    async def handle_info(self, msg):
-        await msg.reply(embed=constants.INFO_REPLY)
-        return
-
-    async def on_message(self, msg: discord.Message):
-        if msg.author.bot:
-            return
-        logger.info("Message received from %s", msg.author.name)
-        if msg.content.startswith("best") or msg.content.startswith("aoc"):
-            await self.handle_best(msg)
-        elif msg.content.startswith("submit"):
-            await self.handle_submit(msg)
-        elif msg.content == "info":
-            await self.handle_info(msg)
-        else:
-            await self.handle_help(msg)
-        return
+        (times1, times2) = lib.get_best_times(day)
+        times1_str = await format_times(times1)
+        times2_str = await format_times(times2)
+        embed = discord.Embed(
+            title=f"Top 10 fastest toboggans for day {day}", color=0xE84611
+        )
+        if times1_str and (part is None or part == 1):
+            embed.add_field(name="Part 1", value=times1_str, inline=True)
+        if times2_str and (part is None or part == 2):
+            embed.add_field(name="Part 2", value=times2_str, inline=True)
+        await ctx.reply(embed=embed)
+
+    # i intentionally did not have the default behavior of automatically choosing part 1 because that's confusing
+    @commands.hybrid_command()
+    async def submit(self, ctx: commands.Context, day: commands.Range[int, 1, 25], part: typing.Literal[1, 2],
+                     code: discord.Attachment):
+        if day > lib.today():
+            raise commands.BadArgument(f"Day {day} is in the future!")
+        logger.info(
+            "Queueing submission for %s, message = [%s], queue length = %s",
+            ctx.author,
+            ctx.args,
+            self.bot.queue.qsize(),
+        )
+        # using a tuple is probably the most readable but shut
+        self.bot.queue.put_nowait((ctx, day, part, await code.read()))
+        await ctx.reply(
+            f"Your submission for day {day} part {part} has been queued. "
+            f"There are {self.bot.queue.qsize()} submissions in the queue."
+        )
+
+    @commands.hybrid_command()
+    async def help(self, ctx: commands.Context):
+        await ctx.reply(embed=constants.HELP_REPLY)
+
+    @commands.hybrid_command()
+    async def info(self, ctx: commands.Context):
+        await ctx.reply(embed=constants.INFO_REPLY)
+
+
+async def prefix(dbot: commands.Bot, message: discord.Message) -> typing.List[str]:
```

Comment:
we use python 3.11 in this house (currently)
so we can use `list[str]` here, no need to use an ugly import
also it is common practice to import things from typing directly, since they are pseudo builtin to typed python